### ISSUE TITLE
Distilling facenet (Hinton, Distilling the Knowledge in a Neural Network)

### ISSUE THREAD

**Issue by speculaas** [davidsandberg/facenet#815]
Dear All,
Anyone tried distillation on David's facenet?

According to : 3 lines in github.com/DushyantaDhyani/kdtf :
soft_targets = teacher_model.predict(batch_x, self.temperature)
self.sess.run([self.train_op, self.merged_summary_op],..., self.soft_Y: soft_targets , ... 
softmax_cross_entropy_with_logits( ..., labels=self.soft_Y )

I am looking to feed inception res v1 's softmax (scaled log probabilities) probabilities into nn4's softmax_cross_entropy_with_logits label.

But, I embarrassingly found out just then the input images and labels are queued.

Anyone know how to enqueue inception res v1 smx probabilities alongside images and labels?

And another question is data_flow_ops is supposed to make things efficient. 

If possible at all, I come up with some solution to enqueue inception res v1 smx probability, would it destroy the original performance gain?

Or instead of dynamically do inference on teacher then feed smx probabilities to student, it's better to do things in 2 passes?

Pass 1, run inception res v1 on all images and store the smx probabilities in a file.
Pass 2, read the teacher smx probabilities from file, then besides the original image and label, also enqueue smx probabilities?

BR,
Jimmy

ps. some related references, study materials : 
https://www.tensorflow.org/api_docs/python/tf/train/batch_join
https://stackoverflow.com/questions/38827264/how-to-use-tensorflow-reader-and-queue-to-read-two-file-at-same-time
https://stackoverflow.com/questions/37581671/how-to-mix-queue-based-and-feed-based-input-in-tensorflow
https://www.tensorflow.org/api_docs/python/tf/FIFOQueue
